### PR TITLE
x11-libs/wxGTK: version bump (3.0.3), use gstreamer:1.0

### PR DESCRIPTION
**Pull request by kuzetsa**
upstream added gstreamer:1.0 support for wxGTK v3.0.3

obsolete patch: [ wxGTK-3.0.2.0-gcc6.patch ] ( gentoo bug #17147 )

patch was an upstream commit and is present as-of v3.0.3:
https://github.com/wxWidgets/wxWidgets/commits/v3.0.3/src/stc/scintilla/src/Editor.cxx

obsolete patch: [ wxGTK-3.0.2.0-webview-fixes.patch ]

specific upstream commit applies the same changes:
https://github.com/wxWidgets/wxWidgets/commit/94b711e9f31645e27f31db1bb107e07eb831f932.patch

slot change / gstreamer references:
gentoo bug #635900 / gentoo bug #601354 / GLSA-201705-10, etc.

Package-Manager: Portage-2.3.8, Repoman-2.3.3

### CODE REVIEW

**Comment by kuzetsa** [comment]
reason for "nomerge" as follows:

things which depend on x11-libs/wxGTK

unexpected behavior spotted during testing:
dev-python/wxpython may need a simple rebuild (due to RDEP)

but it could be possible wxpython needs to track wxGTK version?

**Comment by kuzetsa** [comment]
wxGTK-3.0.2.0 also had a non-gtk3 version, so planning to add (and test) a non-gtk3 revision as well for this 3.0.3 bump PR since that should have compatibility with wxPython and other things which don't seem to work right  with gtk3 enabled.

*The reason the initial commit / bump produced a gtk3 slot is because of being applied against the wxGTK-3.0.2.0-r301 revision rather than -r2 or -r3 ebuild (with those revisions being gtk2)*

**Comment by kuzetsa** [comment]
laptop is no longer affected by GLSA-201705-10 according to:$ glsa-check -t affected

This was an issue due to media packages sometimes having deps like:
x11-libs/wxGTK:${WX_GTK_VER}=[gstreamer,X]

dev-python/wxpython, media-libs/wxsvg, app-arch/p7zip etc. are all working
(*but only work with the non-gtk3, -r1 revision*)

**Comment by kuzetsa** [comment]
**Followup / note:** In the course of removing 2 obsolete patches for this bump, removal of an sed-based fix for [gentoo bug #536004](https://bugs.gentoo.org/536004) wasn't evaluated.

It is not a build-time issue for this package. Specific testing for a regression was not done, though a naive rebuild for locally installed packages (*which depend on wxGTK*) all succeeded without issue, and run normally.

I believe net-ftp/filezilla is the package which will need checked, though I am uncertain which version(s) could be affected, because bug #536004 doesn't specify.

**Comment by titanofold** [comment]
What does the following command give for you:

    /usr/bin/wx-config-3.0 --version

**Comment by kuzetsa** [comment]
@titanofold according to:$ /usr/bin/wx-config-3.0 --version  
3.0.3

File: x11-libs/wxGTK/wxGTK-3.0.3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		x11-libs/cairo[${MULTILIB_USEDEP}]
+		x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+		x11-libs/gdk-pixbuf[${MULTILIB_USEDEP}]
+		x11-libs/libSM[${MULTILIB_USEDEP}]
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXxf86vm[${MULTILIB_USEDEP}]
+		x11-libs/pango[${MULTILIB_USEDEP}]
+		gstreamer? (
+			media-libs/gstreamer:1.0[${MULTILIB_USEDEP}]
+			media-libs/gst-plugins-base:1.0[${MULTILIB_USEDEP}] )
```
Comment (kuzetsa) [inline x11-libs/wxGTK/wxGTK-3.0.3.1.ebuild]: GLSA-201705-10 blocker

File: x11-libs/wxGTK/wxGTK-3.0.3.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+PDEPEND=">=app-eselect/eselect-wxwidgets-20131230"
+
+LICENSE="wxWinLL-3
+		GPL-2
+		doc?	( wxWinFDL-3 )"
+
+S="${WORKDIR}/wxWidgets-${PV}"
+
+src_prepare() {
+	epatch "${FILESDIR}"/${PN}-3.0.0.0-collision.patch
```
Comment (kuzetsa) [inline x11-libs/wxGTK/wxGTK-3.0.3.1.ebuild]: unclear if collision patch is working correctly.

eselect module possibly needs updated as well

**Comment by kuzetsa** [comment]
**not mention in commit message:**  
updated URL for upstream package source.

**Comment by kuzetsa** [comment]
Should've originally been linked to:
https://bugs.gentoo.org/629208

---

**closing PR:**  
will coordinate with wxwidgets project directly;  
*wxGTK maintainer(s) not using github for bumps*

**Pull request closed without merging**